users/user-guide-query/multi-stage-query/operator-types/hash_join.md

```diff
@@ -0,0 +1,210 @@
+---
+description: >-
+  Describes the hash join relation operator in the multi-stage query engine.
+---
+
+# Hash join relational operator
+
+The hash join operator is used to join two relations using a hash join algorithm. 
+It is a binary operator that takes two inputs, the left and right relations, and produces a single output relation. 
+
+This is the only join operator in the multi-stage query engine and it is always created as a result of a query that 
```

Comment:
with lookup join support, does this sentence need to be changed?

Reply:
probably. And being honest, some joins can be optimized as semi-joins (and therefore pipeline breakers) as well.